employee-rostering-webapp/src/test/java/org/optaweb/employeerostering/webapp/rotation/RotationRestServiceIT.java

```diff
@@ -144,34 +171,28 @@ public void testProvisionAndPublish() {
         ShiftTemplateView shiftTemplateC = createShiftTemplate(spotA, null, Duration.ofDays(1), Duration.ofHours(9));
         ShiftTemplateView shiftTemplateD = createShiftTemplate(spotB, employeeA, Duration.ofDays(1), Duration.ofHours(9));
 
-        Map<Long, List<ShiftTemplateView>> spotIdToShiftTemplateViewListMap = new HashMap<>();
-        spotIdToShiftTemplateViewListMap.put(spotA.getId(), Arrays.asList(shiftTemplateA, shiftTemplateC));
-        spotIdToShiftTemplateViewListMap.put(spotB.getId(), Arrays.asList(shiftTemplateB, shiftTemplateD));
+        rotationRestService.addShiftTemplate(TENANT_ID, shiftTemplateA);
+        rotationRestService.addShiftTemplate(TENANT_ID, shiftTemplateB);
+        rotationRestService.addShiftTemplate(TENANT_ID, shiftTemplateC);
+        rotationRestService.addShiftTemplate(TENANT_ID, shiftTemplateD);
 
-        RotationView rotationView = new RotationView();
-        rotationView.setTenantId(TENANT_ID);
-        rotationView.setSpotList(Arrays.asList(spotA, spotB));
-        rotationView.setEmployeeList(Arrays.asList(employeeA, employeeB));
-        rotationView.setSpotIdToShiftTemplateViewListMap(spotIdToShiftTemplateViewListMap);
-        rotationView.setRotationLength(2);
-        shiftRestService.updateRotation(TENANT_ID, rotationView);
-        assertClientResponseEmpty();
+        assertClientResponseOk();
 
         RosterState oldRosterState = rosterRestService.getRosterState(TENANT_ID);
         assertClientResponseOk();
         PublishResult publishResult = rosterRestService.publishAndProvision(TENANT_ID);
         assertClientResponseOk();
         RosterState newRosterState = rosterRestService.getRosterState(TENANT_ID);
         assertClientResponseOk();
-        List<ShiftView> shiftList = shiftRestService.getShifts(TENANT_ID);
+        List<ShiftView> shiftList = shiftRestService.getShiftList(TENANT_ID);
 
         assertThat(publishResult.getPublishedFromDate()).isEqualTo(oldRosterState.getFirstDraftDate());
         assertThat(publishResult.getPublishedToDate()).isEqualTo(oldRosterState.getFirstDraftDate().plusDays(oldRosterState.getPublishLength()));
 
         assertThat(newRosterState.getFirstDraftDate()).isEqualTo(oldRosterState.getFirstDraftDate().plusDays(oldRosterState.getPublishLength()));
         assertThat(newRosterState.getDraftLength()).isEqualTo(oldRosterState.getDraftLength());
-        assertThat(newRosterState.getUnplannedRotationOffset()).isEqualTo(oldRosterState.getPublishLength() % 2);
+        assertThat(newRosterState.getUnplannedRotationOffset()).isEqualTo(oldRosterState.getPublishLength() % oldRosterState.getRotationLength());
         assertThat(newRosterState.getPublishDeadline()).isEqualTo(oldRosterState.getPublishDeadline().plusDays(oldRosterState.getPublishLength()));
-        assertThat(shiftList).size().isEqualTo(oldRosterState.getPublishLength() * 2);
+        assertThat(shiftList).size().usingComparator(Comparator.naturalOrder()).isEqualTo(4);
```

Comment:
+1. Here `size()` is not `Collection`'s size method but `ListAssert`'s size method that is used to perform advanced assertions on the size (e.g. `isGreaterThan()`).

This should probably do the job in this case:

```java
assertThat(shiftList).hasSize(4);
```

Reply:
Will change (in case you are wondering why I used a comparator, I encounter a situation where identical integers in value were seen as different (since they were different objects)).